src/main/java/com/google/api/codegen/advising/AdviserRule.java

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2017 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.advising;
+
+import com.google.api.codegen.ConfigProto;
+import com.google.api.tools.framework.model.Model;
+import java.util.List;
+
+public interface AdviserRule {
+  String getName();
```

Comment:
Explain the syntax of the rule name with examples?

Reply:
I meant can we control the granularity for proto element?

Reply:
How would we specify that on the command line?

Reply:
rule-name@element-name (e.g. FooRule@example.Bar,example.Bak)

Reply:
Ok, that functionality isn't there yet, but I think I can add it in.

Reply:
You can extend it in your followup PR.